framework/operator.go

```diff
@@ -2,8 +2,55 @@ package framework
 
 import (
 	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
 )
 
+// Config represents the configuration used to create a new operator framework.
+type Config struct {
+	// Dependencies.
+	Logger micrologger.Logger
+}
+
+// DefaultConfig provides a default configuration to create a new operator
+// framework by best effort.
+func DefaultConfig() Config {
+	var err error
+
+	var newLogger micrologger.Logger
+	{
+		config := micrologger.DefaultConfig()
+		newLogger, err = micrologger.New(config)
+		if err != nil {
+			panic(err)
+		}
+	}
```

Comment:
I noticed recently this is inconsistent in our codebase.
Sometimes we provide a default logger, sometimes not.

I was thinking about DefaultConfig functions. After some experiments I thought they should provide default settings, but not default dependencies. I.e. logger should be nil in this case.

It's not a blocker. Let's discuss and make it eventually consistent :D